lms/static/scripts/frontend_apps/utils/test/use-warn-on-page-unload-test.js

```diff
@@ -18,43 +18,53 @@ describe('useWarnOnPageUnload', () => {
   };
   const createComponent = () => mount(<FakeComponent />);
 
+  const registerBeforeUnloadListenerAsPromise = () =>
```

Comment:
Maybe `waitForBeforeUnloadEvent`?

Reply:
Actually, never mind. With my second comment above I realized I can combine both helper functions and call it as you proposed.